arch/Kconfig

```diff
@@ -540,6 +545,12 @@ config ARCH_HAS_COHERENCE
 config ARCH_HAS_THREAD_LOCAL_STORAGE
 	bool
 
+config ARCH_HAS_PICOLIBC
+	bool
+	help
+	  Selected when the target architecture has validated support
+	  for picolibc.
```

Comment:
IMHO, this would better handled via `depends on ARC || ARM || ARM64 || ...` from the `config PICOLIBC` because `ARCH_HAS_*` are for the availability of the architecture/ABI-specific kernel features, but no strong opinion in this particular instance since the picolibc will soon become an integral part of Zephyr.

Reply:
Hrm. What I want is a way to know when the picolibc-specific tests can be enabled; this replaced a pile of architecture tests in the testcase.yaml files. I guess I could stick a config PICOLIBC_SUPPORTS_ARCH in lib/libc/Kconfig? Name suggestions welcome, just want to make sure we have a single place which determines whether picolibc is supported or not, instead of littering them around the codebase.

Reply:
> What I want is a way to know when the picolibc-specific tests can be enabled

In that case, adding `PICOLIBC_SUPPORTED` or similar that has `default y` and `depends on ARC || ARM || ...` in `lib/libc/Kconfig` should do.

Also `config PICOLIBC` should depend on `PICOLIBC_SUPPORTED` since it would not be reasonable to allow users to select the picolibc for the archs that are not tested.

One question I have is: do we really need this? Is picolibc known to not work with any of the architectures currently supported in Zephyr? (i.e. do any of the tests fail on them?)

Reply:
Yeah, that looks better -- stick 'config ARCH_HAS_PICOLIBC' in lib/libc/Kconfig. Let me know if there's a better name for this.

Reply:
Since `config ARCH_HAS_PICOLIBC` is no longer in `arch/Kconfig`, `PICOLIBC_SUPPORTED` or `PICOLIBC_SUPPORTS_ARCH` would be better.

Though, this symbol may very well be generalised to indicate whether the picolibc with the current configuration is supported in general (i.e. not just for the different archs, but also for toolchains); in that sense, `PICOLIBC_SUPPORTED` could be a better choice than `PICOLIBC_SUPPORTS_ARCH`; but for now, either would be ok really.

Reply:
Also still wondering:

> One question I have is: do we really need this? Is picolibc known to not work with any of the architectures currently supported in Zephyr? (i.e. do any of the tests fail on them?)

Reply:
> Since `config ARCH_HAS_PICOLIBC` is no longer in `arch/Kconfig`, `PICOLIBC_SUPPORTED` or `PICOLIBC_SUPPORTS_ARCH` would be better.
> 
> Though, this symbol may very well be generalised to indicate whether the picolibc with the current configuration is supported in general (i.e. not just for the different archs, but also for toolchains); in that sense, `PICOLIBC_SUPPORTED` could be a better choice than `PICOLIBC_SUPPORTS_ARCH`; but for now, either would be ok really.

Sounds like `PICOLIBC_SUPPORTED` would be a better choice. I'll switch.

Reply:
> One question I have is: do we really need this? Is picolibc known to not work with any of the architectures currently supported in Zephyr? (i.e. do any of the tests fail on them?)

I tried x86_64 and there were some unexpected failures which I haven't debugged. Note that I'm doing much more significant testing off of the 'picolibc-fixes' branch, which switches the default C library to picolibc and then runs all of the tests, instead of only the subset included in the picolibc-module test. Small steps.

Reply:
> Note that I'm doing much more significant testing off of the 'picolibc-fixes' branch, which switches the default C library to picolibc and then runs all of the tests, instead of only the subset included in the picolibc-module test. Small steps.

Ok, thanks for clarifying. We can gradually address those issues after this PR is merged.